internal/controllers/machineset/machineset_controller.go

```diff
@@ -747,14 +754,36 @@ func (r *Reconciler) syncReplicas(ctx context.Context, s *scope) (ctrl.Result, e
 // There are small differences in how we calculate the Machine depending on if it
 // is a create or update. Example: for a new Machine we have to calculate a new name,
 // while for an existing Machine we have to use the name of the existing Machine.
-func (r *Reconciler) computeDesiredMachine(machineSet *clusterv1.MachineSet, existingMachine *clusterv1.Machine) *clusterv1.Machine {
+func (r *Reconciler) computeDesiredMachine(ctx context.Context, machineSet *clusterv1.MachineSet, existingMachine *clusterv1.Machine) (*clusterv1.Machine, error) {
+	machineName := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", machineSet.Name))
+	// If the MachineSet is part of a MachineDeployment, check MachineNamingStrategy
+	// and name Machine accordingly.
+	if isDeploymentChild(machineSet) {
+		owner, err := r.getOwnerMachineDeployment(ctx, machineSet)
+		if err != nil {
+			return nil, err
+		}
+		nameTemplate := "{{ .machineDeployment.name }}-{{ .random }}"
+		if owner.Spec.MachineNamingStrategy != nil && owner.Spec.MachineNamingStrategy.Template != "" {
+			nameTemplate = owner.Spec.MachineNamingStrategy.Template
+			if !strings.Contains(nameTemplate, "{{ .random }}") {
```

Comment:
This should be checked on admission and not be a possible path in practice. I think It's worth a // comment clarifying that.

Reply:
we are checking it in webhook: https://github.com/kubernetes-sigs/cluster-api/pull/11172/files#diff-b6bae0ed28a4f55a318d296bc63afcff0c7751968346a551ef23f7360a89fd74R313 
Where should I add the // comment ?

Reply:
I think It'd be nice to have a comment on the piece of code that will never be exercised in practice
e.g.
```
// This should never happen as this is validated on admission.
if !strings.Contains(nameTemplate, "{{ .random }}") {
...
```